NWNXLib/Services/Commands/Commands.cpp

```diff
@@ -0,0 +1,128 @@
+#include "Services/Commands/Commands.hpp"
+#include <stdio.h>
+#include <dlfcn.h>
+
+namespace NWNXLib {
+namespace Services {
+
+static Commands *g_commands;
+
+Commands::Commands()
+{
+    g_commands = this;
+    RegisterCommand("nwnx_commands_test", [](std::string& s){ LOG_NOTICE("Selftest command. Args: '%s'", s.c_str()); });
+}
+
+Commands::~Commands()
+{
+    g_commands = nullptr;
+
+}
+
+bool Commands::RegisterCommand(const std::string& cmd, CommandFunc func)
+{
+    auto it = m_commandMap.find(cmd);
+    if (it == m_commandMap.end())
+    {
+        m_commandMap[cmd] = func;
+        LOG_INFO("Registering command '%s'", cmd.c_str());
+        return true;
+    }
+    else
+    {
+        LOG_WARNING("Command '%s' already registered", cmd.c_str());
+        return false;
+    }
+}
+
+void Commands::UnregisterCommand(const std::string& cmd)
+{
+    LOG_INFO("Unregistering command '%s'", cmd.c_str());
+    m_commandMap.erase(cmd);
+}
+
+bool Commands::ScheduleCommand(const std::string& cmdline)
+{
+    std::string cmd, args;
+    size_t space = cmdline.find(' ');
+
+    if (space == std::string::npos) // No args
+    {
+        cmd = cmdline.substr(0, cmdline.length() - 1);
```

Comment:
Probably a dumb question, but why do you drop the last character from the command if there's no space?

Reply:
Because it returns the darned `\n` as well..